pg_search/src/postgres/customscan/aggregatescan/mod.rs

```diff
@@ -451,6 +604,91 @@ fn extract_aggregates(args: &CreateUpperPathsHookArgs) -> Option<Vec<AggregateTy
     Some(aggregate_types)
 }
 
+/// Identify an aggregate function by its OID and extract field name from its arguments
+unsafe fn identify_aggregate_function(
+    aggref: *mut pg_sys::Aggref,
+    relation_oid: pg_sys::Oid,
+) -> Option<AggregateType> {
+    let aggfnoid = (*aggref).aggfnoid;
+
+    // Get the function name to identify the aggregate
+    let func_name = get_aggregate_function_name(aggfnoid)?;
+
+    // Extract the field name from the first argument
+    let field_name = extract_field_name_from_aggref(aggref, relation_oid);
+
+    match func_name.as_str() {
+        "count" => Some(AggregateType::Count),
+        "sum" => Some(AggregateType::Sum { field: field_name? }),
+        "avg" => Some(AggregateType::Avg { field: field_name? }),
+        "min" => Some(AggregateType::Min { field: field_name? }),
+        "max" => Some(AggregateType::Max { field: field_name? }),
+        _ => {
+            pgrx::debug1!("Unsupported aggregate function: {}", func_name);
+            None
+        }
+    }
+}
+
+/// Get the name of an aggregate function from its OID
+unsafe fn get_aggregate_function_name(aggfnoid: pg_sys::Oid) -> Option<String> {
+    // Use well-known PostgreSQL function OIDs for standard aggregates
+    // These are consistent across PostgreSQL versions
+    match aggfnoid.to_u32() {
```

Comment:
I have an idea to make this more robust and less fragile (especially with different PG versions) in a follow-up PR.